### PR TITLE
Sound

### LINKED ISSUE #14: [FEATURE] Add Sound
### Reference issues

_No response_

### Summary

Specifically, support `"type": "Sound"` in the project JSON.

### Basic examples

See summary

### Drawbacks

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by thayerh**
## Reference issue

This PR Closes #14 

## Description

Adds a third thread to run continuous sound on. Uses javax.sound.midi library and reimplements some faulty methods from JFugue library.

### CODE REVIEW

File: src/test/ExampleProjects/CatsAndDogs/CatsAndDogs.json [hunk 1]
```diff
@@ -52,7 +52,6 @@
     },
     {
       "comment": "Sound's value is shown in the Other Memory tab.",
-      "comment2": "The original C# MIPS Emulator supports Sound in the type field and plays the sound. Unfortunately, we don't yet :(",
```
Comment (jesse-wei) [inline src/test/ExampleProjects/CatsAndDogs/CatsAndDogs.json:55]: :D

**Comment by jesse-wei** [comment]
> reimplements some faulty methods from JFugue library.

Which JFugue methods are faulty?

**Comment by thayerh** [comment]
The MicrotonePreprocessor is what it uses for configuring tones that don't align with equal temperament (keys on a piano). The functions in that class put the tone down an octave and don't account for differences from equal temperament very well.

**Comment by jesse-wei** [comment]
About to merge this, but attaching an email I had sent Thayer on 9/30/24, 00:16.

Sorry for the delay, but I just tested both branches `sound` and `leds` on Ubuntu 24.04, and they work. Specifically, LED's seem to work perfectly, and they look good in the GUI. The changes in that branch can be merged to the main branch as soon as 2 things are removed, the print statements in the console and the additional new project files in the root directory of the repository (though these could be moved to src/test/ExampleProjects, if you want).

Also, sound seems to work pretty well on my Rubik's project, though it sounds different from my demo, and I think this could be due to execution speed being different between the board and the emulator. For reference, here's the sound from my [demo video from Nov 27, 2022](https://youtu.be/CWI60TmpJHM?t=167). And here's the [sound from the emulator](https://youtu.be/E8ieUp99EbA) (`sound` branch) on the same assembly code. Specifically, the difference seems to be a factor of ~2 because in my demo video, the jingle lasts for 10 seconds and in the emulator, the jingle lasts for 17 seconds.

Two questions,

1. Do you also think this is due to the execution speed of the emulator and board being different? Or could it be something else (which I think is unlikely)?
2. Do you notice the same issue on your project, which I assume also has sound?

Best,
Jesse

**Comment by jesse-wei** [comment]
@thayerh Can you review my commit 21a3122 that adds support for `"type": "Sound"` in the JSON, rather than mapping it as `DataMemory` with name `"Sound"`?

**Comment by thayerh** [comment]
All of this looks great to me. Definitely cleans up the code.

**Comment by jesse-wei** [comment]
> All of this looks great to me. Definitely cleans up the code.

Thanks, merging this. Thanks again for implementing sound and LED's!

I will create new .jar files and re-release after fixing the execution speed limitation and some other nitpicks.

**Pull request merged**